src/leo_object_storage_haystack.erl

```diff
@@ -305,6 +305,51 @@ head_with_calc_md5(MetaDBId, StorageInfo, Key, MD5Context) ->
         Other -> Other
     end.
 
+%% @doc Retrieve a metadata from backend_db
+%%      AND check if the corresponding AVS is broken.
+%%
+-spec(head_with_check_avs(MetaDBId, StorageInfo, Key, CheckMethod) ->
+             {ok, binary()} |
+             not_found |
+             {error, any()} when MetaDBId::atom(),
+                                 StorageInfo::#backend_info{},
+                                 Key:: binary(),
+                                 CheckMethod::atom()).
+head_with_check_avs(MetaDBId, StorageInfo, Key, check_header) ->
+    case catch leo_backend_db_api:get(MetaDBId, Key) of
+        {ok, MetadataBin} ->
+            Metadata_1 = binary_to_term(MetadataBin),
+            #backend_info{read_handler = ReadHandler} = StorageInfo,
+            HeaderSize = erlang:round(?BLEN_HEADER/8),
+            case leo_file:pread(ReadHandler, Metadata_1#?METADATA.offset, HeaderSize) of
+                {ok, HeaderBin} ->
+                    case leo_object_storage_transformer:header_bin_to_metadata(HeaderBin) of
+                        {error, _Cause} ->
+                            {error, _Cause};
+                        _Metadata ->
```

Comment:
@mocchira I thought it would be better to compare `Metadata_1` with `_Metadata`.

Reply:
@yosukehara I think the complete check (check the all fields one by one) is a bit overspec and consume a bit cpu cycle more than necessary (as header_bin_to_metadata already checks if each filed is valid or not), how about checking only a few fields in #?METADATA like
- addr_id
- ksize

?

Reply:
@mocchira I've understood. [This row](https://github.com/leo-project/leo_object_storage/pull/20/files#diff-5f19f9835375c9b5354f0f744b008963R326) checks the Metadata's format. It is enough to us. This PR will be merged soon.